app/src/main/java/org/mozilla/fenix/browser/FenixSnackbarDelegate.kt

```diff
@@ -13,19 +14,16 @@ class FenixSnackbarDelegate(val view: View, private val anchorView: View?) :
     ContextMenuCandidate.SnackbarDelegate {
     override fun show(
         snackBarParentView: View,
-        text: Int,
+        @StringRes text: Int,
         duration: Int,
-        action: Int,
+        @StringRes action: Int,
         listener: ((v: View) -> Unit)?
     ) {
         val snackbar = FenixSnackbar.make(view, Snackbar.LENGTH_LONG).setText(view.context.getString(text))
-        if (listener != null) {
-            val newListener = {
+        if (listener != null && action != 0) {
+            snackbar.setAction(view.context.getString(action)) {
```

Comment:
In this case, I think it makes more sense to use a single if statement instead of a safe call with an if statement.

```kt
listener?.let {
  if (action != 0) {
    snackbar.setAction(view.context.getString(action)) { 
      listener.invoke(view) 
    }
  }
}
```
feels harder to read compared to
```kt
if (listener != null action != 0) {
  snackbar.setAction(view.context.getString(action)) { 
    listener.invoke(view) 
  }
}
```